ondemand/README.md

```diff
@@ -913,6 +915,95 @@ Jupyter system app in the menu along with your sandbox development app.
+choices. Sites only have to add more YAML to a `form.yml` to enable this behaviour.  Let's
+see some examples.
+
+### Changing min & max values
+
+Let's put some rules around the debug queue.  We set a static `min` and `max` of 200 and 1000
+respectively.  But in this example, we want different min and max values for the debug queue.
+
+We can configure this behaviour with these `data-min-` and `data-max-` directives attached
+to a given option.  When the `debug` queue is choosen we'll automatically set the min and
+maximum values of the `memory` field.
+
+Note that we're also setting the `compute` min and maxes again. This is currently the only way
+to reset back to defaults if there are any.
+
+```yaml
+# form.yml, only showing custom_queue for brevity.
+  custom_queue:
+    widget: "select"
+    label: "Partition"
+    options:
+      - [
+          "Compute", "compute",
+          data-min-memory: 200,         # set the compute queue back to static defaults
+          data-max-memory: 1000,
+        ]
+      - [
+          "Debug", "debug",
+          data-min-memory: 400,         # change min & max for debug queue
+          data-max-memory: 600,
+        ]
+```
+
+![A gif of a user interacting with the form with the dynamic additons described above. The default queue compute is chosen. The user can range memory form from 200 to 1000. The user chooses the debug queue. The user can now range the memory form item from 400 to 600. Switching back to the compute queue and the user can again range the memory form item from 200 to 1000.](imgs/dynamic_min_max.gif)
+
+### Changing values
+
+Let's take this a little further.  Now, when we choose `compute` or `debug` queue, let's automatically
+set the Slurm account we want to use.  Note we'll need to add `bc_account` back, as it's what we'll be
+setting.
+
+We can add the `data-set` directives on the same `custom_queue` form options.  When users choose the
+`debug` queue we'll automatically set the account to `staff`.  When we choose the `compute` queue we
+will set the `sfoster` account.
+
+```yaml
+# form.yml, only showing custom_queue for brevity.
+attributes:
+  custom_queue:
+    widget: "select"
+    label: "Partition"
+    options:
+      - [
+          "Compute", "compute",
+          data-min-memory: 200,
+          data-max-memory: 1000,
+
+          data-set-bc-account: 'sfoster'    # set the account to sfoster when using compute
+        ]
+      - [
+          "Debug", "debug",
+          data-min-memory: 400,
+          data-max-memory: 600,
+
+          data-set-bc-account: 'staff'      # set the account to staff when using debug
+        ]
+form:
+  - bc_account
+```
+
+To use the `sfoster` account you need to run these commands to add the `hpcadmin` user to
+that account.
+
+```
+sacctmgr add user hpcadmin account=sfoster
+sacctmgr modify user where user=hpcadmin set defaultaccount=staff
+```
```

Comment:
Curious if you can just use the sfoster account for the tutorial? Why use the hpcadmin account at all? The idea for the sfoster account was to be setup like a typical PI at an hpc center. The idea for the hpcadmin account was to be like a sysadmin. 

Either way, the hpcadmin slurm association should already be set with the default account of staff.

Reply:
Yea I could flip it, but really what I need is a user with 2 accounts available to it.

Reply:
We can create whatever slurm associations you need in the slurm entrypoint when it first boots up. What if we modified the default slurm associations [here](https://github.com/ubccr/hpc-toolset-tutorial/blob/master/slurm/entrypoint.sh#L108) to be this: 

```
 sacctmgr -i add account staff Cluster=hpc Description=staff
 sacctmgr -i add user hpcadmin DefaultAccount=staff AdminLevel=Admin
 sacctmgr -i add account sfoster Cluster=hpc Description="PI account sfoster"
 sacctmgr -i add user sfoster DefaultAccount=sfoster
 sacctmgr -i add user astewart DefaultAccount=sfoster
 sacctmgr -i add account cgray Cluster=hpc Description="PI account cgray"
 sacctmgr add user sfoster Account=cgray
```

That will create default slurm associations like the following: 

```
[hpcadmin@frontend ~]$ sacctmgr list associations cluster=hpc format=Account,Cluster,User,Fairshare tree
Account                 Cluster       User     Share 
-------------------- ---------- ---------- --------- 
root                        hpc                    1 
 root                       hpc       root         1 
 cgray                      hpc                    1 
  cgray                     hpc    sfoster         1 
 sfoster                    hpc                    1 
  sfoster                   hpc   astewart         1 
  sfoster                   hpc    sfoster         1 
 staff                      hpc                    1 
  staff                     hpc   hpcadmin         1 
```

And you can use the `sfoster` user for the tutorial with accounts `sfoster` or `cgray`: 

```
[hpcadmin@frontend ~]$ sacctmgr show user sfoster -s list format=Account,Cluster,DefaultAccount
   Account    Cluster   Def Acct 
---------- ---------- ---------- 
     cgray        hpc    sfoster 
   sfoster        hpc    sfoster 
```

Would this work?

Reply:
Please don't create associations for all of these as that breaks the coldfront part of the demo.  We need 'cgray' to have no account or association to start out with.

Reply:
@dsajdak no prob. we can create another account with an arbitrary name.

Reply:
On second thought, we probably used the `hpcadmin` user because we do go through promoting an app through to production and/or have instructions to modify the OOD system.

> we can create another account with an arbitrary name.

This is totally fine for me.

Reply:
@johrstrom sounds good. Perhaps the easiest thing is to just add the hpcadmin user to the sfoster account. So, we should just be able to add this single line. If you want, feel free to add to this PR or I can create a separate one. Whatever works.  

```
 sacctmgr -i add user hpcadmin Account=sfoster
```